cmd/credentialUtil.go

```diff
@@ -69,6 +71,47 @@ func GetUserOAuthTokenManagerInstance() *common.UserOAuthTokenManager {
 	return currentUserOAuthTokenManager
 }
 
+/*
+ * GetInstanceOAuthTokenInfo returns OAuth token, obtained by auto-login,
+ * for current instance of AzCopy.
+ */
+func GetOAuthTokenManagerInstance() (*common.UserOAuthTokenManager, error) {
+	var err error
+	autoOAuth.Do(func() {
+		var lca loginCmdArgs
+		if glcm.GetEnvironmentVariable(common.EEnvironmentVariable.AutoLoginType()) == "" {
+			err = errors.New("no login type specified")
+			return
+		}
+		// Fill up lca
```

Comment:
I think it would be safe to assume off environment variables (such as ApplicationID, CertificatePath,  or something else. Note that no environment variables are shared between SPN, MSI, or device (which uses none anyway).